hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/checksum/ContainerChecksumTreeManager.java

```diff
@@ -378,6 +380,21 @@ public ByteString getContainerChecksumInfo(KeyValueContainerData data) throws IO
     }
   }
 
+  public static Optional<ContainerProtos.ContainerChecksumInfo> readChecksumInfo(KeyValueContainerData data) {
```

Comment:
In #7474 the `read` method is made public. That method is non-static but the only field it uses is the metrics. Can we make the public `read` call this method but wrap it in the metrics update to reduce duplication?

Reply:
Yes we can do this, I'll keep this comment open. Depending on which PR merges first I'll update the other PR.